test/topicFilterCopy.spec.ts

```diff
@@ -0,0 +1,186 @@
+  const promises = new Array();
+
+  // Now we can start sending orders.
+  promises.push(sendOrder({ Color: "blue", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 10, Priority: "high" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "blue", Quantity: 10, Priority: "low" }));
+  promises.push(sendOrder({ Color: "blue", Quantity: 5, Priority: "high" }));
+  promises.push(sendOrder({ Color: "blue", Quantity: 10, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 10, Priority: "low" }));
+  promises.push(sendOrder({ Color: "red", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 10, Priority: "high" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 5, Priority: "low" }));
+  promises.push(sendOrder({ Color: "yellow", Quantity: 10, Priority: "low" }));
+
+  // wait until all the send tasks are complete
+  await Promise.all(promises);
+}
+
+async function receiveOrders(client: SubscriptionClient): Promise<ServiceBusMessage[]> {
+  const receivedMsgs: ServiceBusMessage[] = [];
+  const receiveListener = client.receive(
+    (msg: ServiceBusMessage) => {
+      receivedMsgs.push(msg);
+      return Promise.resolve();
+    },
+    (err: Error) => {
+      should.not.exist(err);
+    }
+  );
+
+  await delay(3000);
+  await receiveListener.stop();
+
+  return receivedMsgs;
+}
+
+describe.only("Boolean filters", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function addFileterAndReceiveOrders(
+    bool: boolean,
+    client: SubscriptionClient
+  ): Promise<ServiceBusMessage[]> {
+    await subscriptionClient.addRule("BooleanFilter", bool);
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1);
+    should.equal(rules[0].name, "BooleanFilter");
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(client);
+
+    return receivedMsgs;
+  }
+
+  it("Subscription with default filter receives all messages", async function(): Promise<void> {
+    const receivedMsgs = await addFileterAndReceiveOrders(false, defaultSubscriptionClient);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 12);
+
+    await testPeekMsgsLength(defaultSubscriptionClient, 0);
+  });
+
+  it("Subscription with false boolean filter does not receive any messages", async function(): Promise<
+    void
+  > {
+    const receivedMsgs = await addFileterAndReceiveOrders(false, subscriptionClient);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 0);
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+
+    await receiveOrders(defaultSubscriptionClient);
```

Comment:
Why are we receiving from default subscription client here?

Reply:
Calling `receiveOrders(defaultSubscriptionClient);` just for the cleanup.